### PR TITLE
add `Result<T>` type alias for `std::result::Result<T, tonic::Status>`

### PR DESCRIPTION
**Pull request by stphnsmpsn**
## Motivation

#1084 

## Solution

Added a simple alias in `lib.rs`: `pub type Result<T, E = Status> = std::result::Result<T, E>;`

### CODE REVIEW

File: tonic/src/lib.rs [hunk 1]
```diff
@@ -122,3 +122,5 @@ pub(crate) type Error = Box<dyn std::error::Error + Send + Sync>;
 #[cfg(feature = "codegen")]
 #[cfg_attr(docsrs, doc(cfg(feature = "codegen")))]
 pub mod codegen;
+
+pub type Result<T> = std::result::Result<T, Status>;
```
Comment (davidpdrsn) [inline tonic/src/lib.rs:126]: This way users can still override the error type if they want.

```suggestion
pub type Result<T, E = Status> = std::result::Result<T, E>;
```

Comment (davidpdrsn) [reply tonic/src/lib.rs:126]: Also missing docs.

Comment (LucioFranco) [reply tonic/src/lib.rs:126]: TIL

**Review by LucioFranco — APPROVED** [review]
LGTM thanks!

**Comment by LucioFranco** [comment]
One cI failure though that is probably cuz of formatting https://github.com/hyperium/tonic/actions/runs/3116115959/jobs/5054123932

**Comment by stphnsmpsn** [comment]
@LucioFranco , it was a formatting issue my bad. Passes `cargo fmt` now.

**Pull request merged**